tests/integration/test_dev_server.py

```diff
@@ -55,75 +70,108 @@ def test_serves_root_dir(site, expected_text) -> None:
                 # before burning the bridge:
-                loop.call_soon_threadsafe(lambda: loop.call_later(0.05, lambda: loop.stop()))
-        await loop.run_in_executor(None, run_test)
+                loop_for_this_test.call_soon_threadsafe(lambda: loop_for_this_test.call_later(0.05, lambda: loop_for_this_test.stop()))
+
+        await loop_for_this_test.run_in_executor(None, run_test)
 
     # We defeat the nikola site building functionality, so this does not actually get called.
-    # But the setup code sets this up and needs a command list for that.
+    # But the code setting up site building wants a command list:
     command_auto.nikola_cmd = ["echo"]
 
     # Defeat the site building functionality, and instead insert the test:
     command_auto.run_initial_rebuild = grab_loop_and_run_test
 
-    # Start the development server
-    # which under the hood runs our test when trying to build the site:
-    command_auto.execute(options=options)
-
-    # Verify the test succeeded:
-    if test_inner_error is not None:
-        raise test_inner_error
-    assert test_was_successful, test_problem_description
-
-
-class MyFakeSite(FakeSite):
-    def __init__(self, config, configuration_filename):
-        self.configured = True
-        self.debug = True
-        self.THEMES = []
-        self._plugin_places = []
-        self.registered_auto_watched_folders = set()
-        self.config = config
-        self.configuration_filename = configuration_filename
-
-
-@pytest.fixture(scope="module")
-def site() -> MyFakeSite:
+    try:
+        # Start the development server
+        # which under the hood runs our test when trying to build the site:
+        command_auto.execute(options=options)
+
+        # Verify the test succeeded:
+        if test_inner_error is not None:
+            raise test_inner_error
+        assert test_was_successful, test_problem_description
+    finally:
+        # Nikola is written with the assumption that it can
+        # create the event loop at will without ever cleaning it up.
+        # As this tests runs several times in succession,
+        # that assumption becomes a problem.
+        LOGGER.info("Cleaning up loop.")
+        # Loop cleanup:
+        assert loop_for_this_test is not None
+        assert not loop_for_this_test.is_running()
+        loop_for_this_test.close()
+        asyncio.set_event_loop(None)
+        # We would like to leave it at that,
+        # but doing so causes the next test to fail.
+        #
+        # We did not find asyncio - API to reset the loop
+        # to "back to square one, as if just freshly started".
+        #
+        # The following code does not feel right, it's a kludge,
+        # but it apparently works for now:
+        if sys.platform == 'win32':
+            # For this case, the auto module has special code
+            # (at module load time! 😟) which we reluctantly reproduce here:
```

Comment:
This was required to work around a bug in asyncio, IIRC it prevented shutting down the server with ^C. You may move it to some later point than loading the module.